### PR TITLE
[hailtop.utils]test for file none

### PR DESCRIPTION
**Pull request by CDiaz96**

### CODE REVIEW

**Review by jigold — DISMISSED** [review]

File: hail/python/test/hailtop/utils/test_utils.py [hunk 1]
```diff
@@ -200,3 +200,13 @@ def test_digits_needed():
...
+
+
+def test_filter_none():
+    assert filter_none([]) == []
+    assert filter_none([None, []]) == [[]]
+    assert filter_none([1, 2, [None]]) == [1, 2, [None]]
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
+    assert filter_none([1, 2, 3.0, None, 5]) == [1, 2, 3.0, 5]
+    assert filter_none(['a', 'b', 'c', None]) == ['a', 'b', 'c']
```
[the remark this answers is not in the document]
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:212]: Added line

File: hail/python/test/hailtop/utils/test_utils.py [hunk 2]
```diff
@@ -200,3 +200,13 @@ def test_digits_needed():
     assert digits_needed(100) == 3
     assert digits_needed(3000) == 4
     assert digits_needed(50000) == 5
+
+
+def test_filter_none():
+    assert filter_none([]) == []
+    assert filter_none([None, []]) == [[]]
+    assert filter_none([1, 2, [None]]) == [1, 2, [None]]
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
```
[the remark this answers is not in the document]
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:209]: Added test

File: hail/python/test/hailtop/utils/test_utils.py [hunk 3]
```diff
@@ -200,3 +200,13 @@ def test_digits_needed():
     assert digits_needed(100) == 3
     assert digits_needed(3000) == 4
     assert digits_needed(50000) == 5
+
+
+def test_filter_none():
+    assert filter_none([]) == []
+    assert filter_none([None, []]) == [[]]
+    assert filter_none([1, 2, [None]]) == [1, 2, [None]]
```
[the remark this answers is not in the document]
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:208]: Added test

File: hail/python/test/hailtop/utils/test_utils.py [hunk 4]
```diff
@@ -200,3 +200,13 @@ def test_digits_needed():
...
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
```
[the remark this answers is not in the document]
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:210]: Removed duplicate

File: hail/python/test/hailtop/utils/test_utils.py [hunk 5]
```diff
@@ -200,3 +200,14 @@ def test_digits_needed():
...
+
+def test_filter_none():
+    assert filter_none([]) == []
+    assert filter_none([None, []]) == [[]]
+    assert filter_none([0, []]) == [0, []]
+    assert filter_none([1, 2, [None]]) == [1, 2, [None]]
+    assert filter_none([1, 3.5, 2, 4,]) == [1, 3.5, 2, 4]
+    assert filter_none([1, 2, 3.0, None, 5]) == [1, 2, 3.0, 5]
+    assert filter_none(['a', 'b', 'c', None]) == ['a', 'b', 'c']
+    assert filter_none([None, [None, [None, [None]]]]) == [[None, [None, [None]]]]
```
Comment (danking) [inline hail/python/test/hailtop/utils/test_utils.py:213]: I like this test, it's a tricky one!

**Review by danking — APPROVED** [review]

**Pull request merged**